products/jbrowse-web/src/JBrowse.js

```diff
@@ -89,7 +103,10 @@ const JBrowse = observer(({ pluginManager }) => {
   useEffect(() => {
     onSnapshot(rootModel, async snapshot => {
       if (adminMode) {
-        const payload = { adminKey: adminKeyParam, config: snapshot.jbrowse }
+        const config = JSON.parse(JSON.stringify(snapshot.jbrowse))
+        revertUris(config)
```

Comment:
what is revertUris purpose?

Reply:
To not overwrite the value of `uri` in the config file on disk with the relative one that is being used

Reply:
Oh I see...for when writing the config back to the server. What if, instead of keeping originalUri, we store baseUri in the file location, and then we can use baseUri+uri to compose a URI? Then baseUri does not get serialized back to the config

Reply:
Do you have a strong preference for that? This is working as is

Reply:
I might belive that baseUri might be a bit preferable. I made an outline for how it might look here https://github.com/GMOD/jbrowse-components/tree/1147_relative_urls_baseuri

If it could, in adminMode or on snapshotting this config, not serialize the baseUri I think that would be ideal, haven't tested that here